packages/framework/src/Model/Transport/TransportRepository.php

```diff
@@ -170,7 +170,8 @@ public function getAllWithEagerLoadedDomainsAndTranslations(
 
         if ($totalWeight !== null) {
             $queryBuilder
-                ->andWhere('t.maxWeight IS NULL OR t.maxWeight >= :maxWeight')
+                ->join('t.prices', 'tp')
+                ->andWhere('tp.maxWeight IS NULL OR tp.maxWeight >= :maxWeight')
```

Comment:
what if we have weights that meets both conditions?

Reply:
hm, interesting question, do you think the entry would be duplicated in the result? The `TransportPrice` is not included in the `SELECT` so IMHO it should be fine, or?